dynamiqs/utils/utils.py

```diff
@@ -735,3 +735,24 @@ def mpow(x: Tensor, n: int) -> Tensor:
                 [0.+0.j, 1.+0.j]])
     """
     return torch.linalg.matrix_power(x, n)
+
+def entropy_vn(x: Tensor) -> Tensor:
+    """Von-Neumann entropy of density matrix
+
+    Args:
+        x _(..., n, n)_: Square matrix.
+
+    Returns:
+        _(...)_ Von_Neumann entropy of x
+
+    Examples:
+        >>> rho = 0.5*dq.fock_dm(2,0) + 0.5*dq.fock_dm(2,1)
+        >>> entropy_vn(rho)
+        tensor(0,6931471806)
+    """
+    if isket(x):
+        x = todm(x)
+    vals = torch.linalg.eigvalsh(x)
+    nzvals = vals[vals != 0]
+    logvals = nzvals.log()
+    return -(nzvals @ logvals).real
```

Comment:
I think this would be more clear (closer to the mathematical definition):
```suggestion
    return -(nzvals * logvals).sum().real
```

Reply:
It is twice as slow
<img width="726" alt="image" src="https://github.com/dynamiqs/dynamiqs/assets/4300678/f0594b24-14d5-4c8e-ac44-9b6080e8d345">

Reply:
Ouw, nice, thanks for the benchmark. Actually there's an issue currently with batched matrix:
```
rho = torch.stack([
    dq.unit(dq.fock_dm(2,0) + dq.fock_dm(2,1),
    dq.unit(dq.fock_dm(2,0) + dq.fock_dm(2,1),
])
dq.utils.entropy_vn(rho)
```
(which is not fixed by my version of the code either)